tools/config.py

```diff
@@ -455,6 +463,49 @@ def add_config_files(self, flist):
                        self.lib_config_data[cfg["name"]]["__config_path"]))
             self.lib_config_data[cfg["name"]] = cfg
 
+    @property
+    def has_regions(self):
+        """Does this config have regions defined?"""
+        if 'target_overrides' in self.app_config_data:
+            target_overrides = self.app_config_data['target_overrides'].get(self.target.name, {})
+            return ('target.bootloader_exec' in target_overrides or
+                    'target.restrict_size' in target_overrides)
+        else:
+            return False
+
+    @property
+    def regions(self):
+        """Generate a list of regions from the config"""
+        cmsis_part = Cache(False, False).index[self.target.device_name]
+        start = 0
+        target_overrides = self.app_config_data['target_overrides'].get(self.target.name, {})
+        try:
+            size = int(cmsis_part['memory']['IROM1']['size'], 0)
+            rom_start = int(cmsis_part['memory']['IROM1']['start'], 0)
+        except KeyError:
+            raise ConfigException("Not enough information in CMSIS packs to "
+                                  "build a bootloader project")
+        if 'target.bootloader_exec' in target_overrides:
+            filename = target_overrides['target.bootloader_exec']
+            _, inteltype = splitext(filename)
+            ih = IntelHex()
+            ih.fromfile(filename, format=inteltype[1:])
+            if ih.minaddr() != 0 and ih.minaddr() != rom_start:
```

Comment:
In the case of .bin files, there is no specified start address, so it defaults to zero.